app/resources/v1/streets.js

```diff
@@ -35,168 +35,189 @@ exports.post = function (req, res) {
     street.creator_ip = requestIp(req)
   }
 
-  const handleCreateStreet = function (err, s) {
-    if (err) {
+  const makeNamespacedId = async function () {
+    try {
+      if (street.creator_id) {
+        const row = await User.findByIdAndUpdate(street.creator_id,
+          { $inc: { 'last_street_id': 1 } },
+          { new: true, upsert: true })
+        const namespacedId = (row) ? row.last_street_id : null
+        if (!namespacedId) {
+          throw new Error(ERRORS.INTERNAL_ERROR)
+        }
+        return namespacedId
+      } else {
+        const row = await Sequence.findByIdAndUpdate('streets',
+          { $inc: { 'seq': 1 } },
+          { new: true, upsert: true })
+        const namespacedId = (row) ? row.seq : null
+        if (!namespacedId) {
+          throw new Error(ERRORS.INTERNAL_ERROR)
+        }
+        return namespacedId
+      }
+    } catch (err) {
       logger.error(err)
-      res.status(500).send('Could not create street.')
-      return
+      throw new Error(ERRORS.INTERNAL_ERROR)
     }
+  } // END function - makeNamespacedId
```

Comment:
So, a better way to do this might be like this:

```js
const makeNamespacedId = async function () {
  let namespacedId

  try {
    if (street.creator_id) {
      namespacedId = await User.findByIdAndUpdate() /* pseudocode */
    } else {
      namespacedId = await Sequence.findByIdAndUpdate()
    }
  } catch (error) {
    // Why here? Because we don't need to try/catch the entire
    // body of the function, just want to catch database errors.
    // We log the error to our logs, then throw an error we can handle.
    logger.error(error)
    throw new Error(ERRORS.CANNOT_CREATE_STREET)
  }

  // Only one check for missing id
  // throws the same error.
  if (!namespacedId) {
    throw new Error(ERRORS.CANNOT_CREATE_STREET)
  }

  return nameSpacedId
} // END function - makeNamespacedId

What's different here? I tried to refactor this function to have less repetition, and a clearer path of logic.

One thing to keep in mind generally is that you do not need to try/catch the entire body of functions. We only want to catch database errors. Try/catching the entire function actually makes it more complex and harder to test obvious errors!